redwood-dom-testing/src/commonMain/kotlin/app/cash/redwood/dom/testing/DomSnapshotter.kt

```diff
@@ -55,22 +57,77 @@ public class DomSnapshotter {
 
     try {
       val boundingClientRect = wrapper.getBoundingClientRect()
-      return DomSnapshot(
-        image = HtmlToImage.toBlob(
-          element = element,
-          options = Options().apply {
-            this.width = ceil(boundingClientRect.width).toInt() - (2 * framingBorderSize)
-            this.height = ceil(boundingClientRect.height).toInt() - (2 * framingBorderSize)
-            this.canvasWidth = this.width
-            this.canvasHeight = this.height
-            this.pixelRatio = frame.pixelRatio
-          },
-        ).await(),
-        framedHtml = wrapper.outerHTML,
-      )
+      val elementWidth = ceil(boundingClientRect.width).toInt() - (2 * framingBorderSize)
+      val elementHeight = ceil(boundingClientRect.height).toInt() - (2 * framingBorderSize)
+
+      if (!scrolling) {
+        return DomSnapshot(
+          images = listOf(
+            html2canvas(
+              element = element as HTMLElement,
+              options = Options().apply {
+                this.backgroundColor = null
+                this.width = elementWidth
+                this.height = elementHeight
+                this.windowWidth = this.width
+                this.windowHeight = this.height
+                this.scale = frame.pixelRatio
+              },
+            ).await()?.encodeImage(),
+          ),
+          framedHtml = wrapper.outerHTML,
+        )
+      } else {
+        // Handle scrollable content
+        val scrollableElement = findScrollableElement(element)
+          ?: throw IllegalStateException("No scrollable element found")
+
+        // Calculate total number of pages needed
```

Comment:
style nit: [comments should be sentences](https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html).
```suggestion
        // Handle scrollable content.
        val scrollableElement = findScrollableElement(element)
          ?: throw IllegalStateException("No scrollable element found")

        // Calculate total number of pages needed.
```